Documentation/guides/messages/xa4214.md

```diff
@@ -0,0 +1,9 @@
+# Compiler Warning XA4214
+
+If the same fully qualified C# type name exists in two or more assemblies and
+both C# types inherit from `Java.Lang.Object`, then the build process will only
+generate an Android Callable Wrapper for *one* of them.
```

Comment:
...now that we're digging into this, an obvious question is this: *Why does anybody care?*

Additionally, the description is wrong.

(Also, let's stop using "Android Callable Wrapper" and stick with "Java Callable Wrapper". ;-)

Consider a variation on the previous [repro instructions](
https://github.com/xamarin/xamarin-android/pull/2258/files#r222784089): two assemblies which both declare the "same" C# type:

```csharp
namespace Lib1
{
  public class EmptyClass : Java.Lang.Object
  {
  }
}
```

We thus have two different `Lib1.EmptyClass` types running around.  However, because there is no `[Register]`, then we hit the default "md5-ify everything" approach, which means we *can* emit two separate Java Callable Wrapper types: `md56bf89f81d87b2a3b578f763649ead9ef.EmptyClass` for `Lib1.EmptyClass, Lib1` and `md551eaf02b2ef90dc8aacda5445b93be5c.EmptyClass` for `Lib1.EmptyClass, Lib2`.

We *can* -- *do* -- generate Java Callable Wrappers for *both* types.

Thus, why care that we have two separate "Lib1.EmptyClass" types running around?  What's the *actual* problem?

The actual problem comes in places where an assembly cannot be specified, such as within `.axml` files.  There, if we used `<Lib1.EmptyClass/>`, there is no way to say "use the `EmptyClass` type from Lib1" vs. "use the `EmptyClass` type from Lib2".  It's ambiguous.

However, *fragments* allow avoiding this, e.g. by using `<fragment type="Lib1.EmptyClass, Lib1" />`.  (No idea if anyone actually uses that functionality, but it exists.)  Again, though, if you use `<fragment type="Lib1.EmptyClass" />` -- no assembly -- then *one* of the types is picked.  The first one.

What is the fix?  Change the type names so that they have different C# Namespace.Type names.  The documentation should likewise mention how to fix the warning.